src/main/java/io/anserini/collection/TweetCollection.java

```diff
@@ -67,4 +76,444 @@ public FileSegment createFileSegment(Path p) throws IOException {
+      }
+
+      if (tweetObj.retweeted_status() == null || !tweetObj.retweeted_status().isPresent()) {
+        retweetStatusId = OptionalLong.empty();
+        retweetUserId = OptionalLong.empty();
+        retweetCount = OptionalLong.empty();
+      } else {
+        retweetStatusId = tweetObj.retweeted_status().get().id();
+        if (tweetObj.retweeted_status().get().user() != null && tweetObj.retweeted_status().get().user().isPresent()) {
+          retweetUserId = tweetObj.retweeted_status().get().user().get().id();
+        } else {
+          retweetUserId = OptionalLong.empty();
+        }
+        retweetCount = tweetObj.retweet_count();
+      }
+
+      if (tweetObj.coordinates() == null || !tweetObj.coordinates().isPresent() ||
+              tweetObj.coordinates().get().coordinates() == null ||
+              !tweetObj.coordinates().get().coordinates().isPresent() ||
+              tweetObj.coordinates().get().coordinates().get().size() < 2) {
+        latitude = OptionalDouble.empty();
+        longitude = OptionalDouble.empty();
+      } else {
+        longitude = tweetObj.coordinates().get().coordinates().get().get(0);
+        latitude = tweetObj.coordinates().get().coordinates().get().get(1);
+      }
+
+      if (tweetObj.lang() == null || !tweetObj.lang().isPresent()) {
+        lang = Optional.empty();
+      } else {
+        lang = tweetObj.lang();
+      }
+
+      followersCount = tweetObj.user().followers_count();
+      friendsCount = tweetObj.user().friends_count();
+      statusesCount = tweetObj.user().statuses_count();
+      screenname = tweetObj.user().screen_name();
+
+      if (tweetObj.user().name() != null && tweetObj.user().name().isPresent()) {
+        name = tweetObj.user().name();
+      } else {
+        name = Optional.empty();
+      }
+
+      if (tweetObj.user().profile_image_url() != null && tweetObj.user().profile_image_url().isPresent()) {
+        profile_image_url = tweetObj.user().profile_image_url();
+      } else {
+        profile_image_url = Optional.empty();
+      }
+
+      jsonString = json;
+      jsonObject = tweetObj;
+
+      return true;
+    }
+
+    public Document fromTSV(String tsv) {
+      String[] columns = tsv.split("\t");
+
+      if (columns.length < 4) {
+        System.err.println("error parsing: " + tsv);
+        return null;
+      }
+
+      id = columns[0];
+      idLong = Long.parseLong(columns[0]);
+      screenname = columns[1];
+      createdAt = columns[2];
+
+      StringBuilder b = new StringBuilder();
+      for (int i = 3; i < columns.length; i++) {
+        b.append(columns[i] + " ");
+      }
+      text = b.toString().trim();
+
+      return this;
+    }
+
+    @Override
+    public String id() {
+      return id;
+    }
+
+    @Override
+    public String content() {
+      return text;
+    }
+
+    @Override
+    public boolean indexable() {
+      return true;
+    }
+
+    public long getIdLong() { return idLong; }
```

Comment:
make formatting consistent.

Reply:
Should we use snake case for our variables across all files? Only 'TweetCollection.Document.TweetObject` uses snake case because the field name has to match exactly in the JSON string when parsing using Jackson.

The other place that uses snake case is `WashingtonPostCollection.Document.WashingtonPostObject`. I believe we've used camel case throughout all other places. That's what I'm currently doing.

Reply:
Oh what I mean above was the braces on a separate line:
```
public long getIdLong() {
  return idLong;
}
```
But I think everything should be `CamelCase` except for the Jackson stuff - leave in snake?

Reply:
Nvm, I've figured out a way to use camel case in all cases including in `TweetCollection.Document.TweetObject` and `WashingtonPostCollection.Document.WashingtonPostObject`.